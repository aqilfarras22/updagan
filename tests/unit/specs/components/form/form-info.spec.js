import VeeValidate from 'vee-validate';
import BootstrapVue from 'bootstrap-vue';
import { mount, shallowMount, createLocalVue } from '@vue/test-utils';

import formInfoComponent from '@/components/form/form-info';

const { it, describe, expect } = global;

const testPropsData = {
  formId: '66066679-1eef-480b-86b7-c7745fa4fb87',
  formName: 'e-visit_1',
  approvalStatus: 'NOT_APPROVED',
  verificationStatus: 'VERIFIED',
  verifiedOn: 1516034703,
  approvedOn: 1516034703,
};

const localVue = createLocalVue();
localVue.use(BootstrapVue);
localVue.use(VeeValidate, {
  fieldsBagName: 'veeFields',
});
const vueAclCheckStub = jest
  .fn()
  .mockReturnValue(Promise.resolve({}));

const vueAclChangeStub = jest
  .fn()
  .mockReturnValue(Promise.resolve({}));

describe('components/form/form-info.vue', () => {
  const createMountedWrapper = (options = {}) => mount(formInfoComponent, {
    localVue,
    propsData: { ...testPropsData },
    ...options,
    mocks: {
      $acl: {
        check: vueAclCheckStub,
        change: vueAclChangeStub,
      },
    },
  });
  it('should create instance', () => {
    const wrapper = createMountedWrapper();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it('should able to receive props', () => {
    const wrapper = createMountedWrapper();
    expect(wrapper.props()).toEqual(testPropsData);
  });

  it('should not create if required props items were not passed', () => {
    // const consoleErrorStub = sinon.stub(console, 'error');
    const wrapper = createMountedWrapper({ propsData: {} });
    expect(wrapper.isEmpty()).toBeTruthy();
  });

  it('should display props which were passed', () => {
    const wrapper = createMountedWrapper();
    const formName = wrapper.find({ ref: 'formName' });
    const verificationStatus = wrapper.find({ ref: 'verificationStatus' });
    const approvalStatus = wrapper.find({ ref: 'approvalStatus' });
    expect(formName);
    expect(verificationStatus);
    expect(approvalStatus);
  });

  it('should not display props which were not passed', () => {
    const wrapper = createMountedWrapper({
      propsData: {
        formId: 'formId_prop',
      },
    });
    const { verificationStatus, approvalStatus, formName } = wrapper.vm.$refs;
    expect(verificationStatus).toBe(undefined);
    expect(approvalStatus).toBe(undefined);
    expect(formName).toBe(undefined);
  });

  it('has the expected html structure', () => {
    const { vm } = shallowMount(formInfoComponent, { localVue,
      propsData: { ...testPropsData },
      mocks: {
        $acl: {
          check: vueAclCheckStub,
          change: vueAclChangeStub,
        },
      },
      sync: false });
    expect(vm.$el).toMatchSnapshot();
  });
});
