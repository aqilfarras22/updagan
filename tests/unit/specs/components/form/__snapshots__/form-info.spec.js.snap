// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/form/form-info.vue has the expected html structure 1`] = `
<div>
  <dl>
    <dt>
      Form Name
    </dt>
     
    <dd>
      e-visit 1
    </dd>
  </dl>
   
  <dl>
    <dt>
      Verification Status
    </dt>
     
    <dd>
      <div
        class="VERIFIED"
      >
        Verified
      </div>
       
      <div
        class="date"
      >
        
        ("1970-01-18T13:07:14.703Z")
      
      </div>
    </dd>
  </dl>
   
  <dl>
    <dt>
      Approval Status
    </dt>
     
    <dd>
      <div
        class="NOT_APPROVED"
      >
        Unapproved
      </div>
       
      <!---->
    </dd>
  </dl>
   
  <div
    class="actions"
  >
    <button-cmp-stub
      class="form-action-button"
      size="lg"
      variant="primary"
    >
      Approve
    </button-cmp-stub>
     
    <button-cmp-stub
      class="form-action-button"
      size="lg"
      variant="primary"
    >
      Unverify
    </button-cmp-stub>
  </div>
   
  <formapprovalmodal-stub
    callapi="true"
    formid="66066679-1eef-480b-86b7-c7745fa4fb87"
  />
   
  <formverificationmodal-stub
    callapi="true"
    formid="66066679-1eef-480b-86b7-c7745fa4fb87"
    verified="true"
  />
</div>
`;
